helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -218,6 +217,8 @@ private void reportTopStateExistence(ResourceControllerDataProvider cache, Curre
       // we observed an entire top state handoff process
       reportSingleTopStateHandoff(cache, lastTopStateInstance, currentTopStateInstance,
           resourceName, partition, clusterStatusMonitor, lastPipelineFinishTimestamp);
```

Comment:
I feel like we should modify the if check to just use `if (lastTopStateInstance !=null)` and reuse the `reportSingleTopStateHandoff` function instead by getting rid of the if statement that returns on `topStateInstance == currentTopStateInstance` in `reportSingleTopStateHandoff` function.